dagshub/common/api/repo.py

```diff
@@ -226,6 +231,83 @@ def get_storage_file(self, path: str) -> bytes:
             raise RuntimeError(error_msg)
         return res.content
 
+    def _get_files_in_path(self, path, revision=None, recursive=False) -> List[ContentAPIEntry]:
+        """
+        Walks through the path of the repo, returning non-dir entries
+        """
+
+        dir_queue = [path]
+        files = []
+
+        progress = get_rich_progress(rich.progress.MofNCompleteColumn())
+        task = progress.add_task("Traversing directories...", total=None)
+
+        def step(step_path):
+            res = self.list_path(step_path, revision)
+            for entry in res:
+                if entry.type == "file":
+                    files.append(entry)
+                elif entry.type == "dir" and recursive:
+                    dir_queue.append(entry.path)
+                # TODO: storage
+            progress.update(task, advance=1)
+
+        with progress:
+            while len(dir_queue):
+                query_path = dir_queue.pop(0)
+                step(query_path)
+
+        return files
+
+    def download(
+        self, remote_path, local_path=".", revision=None, recursive=True, keep_source_prefix=True, redownload=False
+    ):
+        """
+        Downloads the contents of the repository at "remote_path" to the "local_path"
+
+        Args:
+            remote_path: path of the folder or folder to download in the repository
+            local_path: where to download the files. Defaults to CWD
+            revision: repo revision, if not specified - uses default repo branch
+            recursive: whether to download files recursively
+            keep_source_prefix: whether to keep the path of the folder in the download path or not
+                example: Given remote_path "src/data" and file "test/file.txt"
+                if True: will download to "<local_path>/src/data/test/file.txt"
+                if False: will download to "<local_path>/test/file.txt"
+            redownload: whether to redownload an already existing file
+                The downloader doesn't do any hash comparisons and only checks if a file already exists in the FS or not
+        """
+        files = self._get_files_in_path(remote_path, revision, recursive)
+        file_tuples = []
+        if local_path is None:
+            local_path = "."
+        local_path = Path(local_path)
+        # Strip the slashes from the beginning so the relative_to logic works
+        if str(remote_path.startswith("/")):
+            remote_path = str(remote_path).lstrip("/")
```

Comment:
Can't you simplify?
Also, why the need to cast to str?
```suggestion
        remote_path = str(remote_path).lstrip("/")
```

Reply:
Good point on simplification, not sure I need the if.
The str cast was defensive - in case remote_path is a Path-like, and not a string